waspc/data/Generator/templates/sdk/wasp/auth/useAuth.ts

```diff
@@ -23,12 +23,14 @@ function createUserGetter(): Query<void, AuthUser | null> {
     try {
       const response = await api.get(getMeRoute.path)
       const userData = superjsonDeserialize<AuthUserData | null>(response.data)
+      // TODO: figure out why overloading is not working
+      // @ts-ignore
```

Comment:
As we discussed IRL: this PR has the goal of unblocking the #2362 PR and nothing will substantially change in the codebase - besides quick fixes and enabling the strict null checking option. It will have the effect that Zod works as expected and all future code has to be written with strict null checks.

I'd go through all the `@ts-ignore`s and do one of the following:
- if we can fix it now - fix it
- if the fix requires going down a rabbit hole - make an issue which we will mention in a comment
- use type assertion instead of `@ts-ignore` if we can - makes the override more precise

Reply:
We fixed this instance by provided an additional overload :)